lib/water_drop/sync_producer.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# WaterDrop library
+module WaterDrop
+  # Sync producer for messages
+  SyncProducer = Class.new(BaseProducer)
+  # Sync producer for messages
+  Producer = SyncProducer
```

Comment:
This is backwards compatibility?

Reply:
No. It is just an alias for convenience. 1.0 is totally not compatible with 0.4

Reply:
I would remove this, we should be strict, either `SyncProducer` or `AsyncProducer` imho.